### DISCUSSION TITLE
Software Load - Stumbling in the blocks

### DISCUSSION THREAD

**Discussion by Alan-Lomax** [DCC-EX/EX-T3-WiFi#75] [Q&A] [answered]
When trying to Load my Firebeetle (for the first time) I am assuming the first step is using the task install the filesystem. (there are two choices one saying (OTA) and one not. Not sure of the difference.

Anyway .. I am getting a message back in the terminal saying error - Wrong boot mode detected (0x13)! The chip needs to be in download mode.  Obviously the connection is good as some feedback is coming through .. also I see some brief messages on the serial monitor during power up.  Thoughts?

**Comment by alex-code** [comment]
Are you on Linux? There was another builder who had some upload trouble on Ubuntu.

Are you able to try other USB ports and cables just to check.

**Reply by Alan-Lomax** [reply]
Not on Linux (Win10) I do have alternate cables and USB ports to try (but don't think that is the issue since I am getting messages in the terminal window) I have also just finished putting together a second firebeetle that I can now test 'outside the case'
Results - different cable, different firebeetle, Different com port (still USB 3) directly on computer all no change.

On a hunch changed to a USB 2 port on a hub .... and it worked !   
Reverted to original firebeetle . Also Success!  
On to the next step!

**Reply by Alan-Lomax** [reply]
I Suspect USB handshaking is at fault here - USB3 asking can you do USB3 speed? Firebeetle says of course - I am USB-C - just as fast as you    But the cable stuck in the middle is USB2 to USBC . and chokes on the USB3 speed !

**Comment by daveclifford** [comment]
The FireBeetle2 has a CH341 USB to Serial Convertor chip that interfaces the signals from your USB A/C socket to the USB-C socket on the FireBeetle2. The cable MUST have two USB data lines to send the serial communications (as well as power/ground).

 I am the Linux user and my issues were: CH341X driver not installed and accessed AND my USB socket was damaged (one of the data lines was broken and the interface would NEVER work!) . After switching to a newer notebook, everything worked fine.

Reading these issues, it appears Windows has the driver installed, and MAC and Linux users need to have the driver installed bwfore uploading the code to the FireBeetle2.

Alex,

I suggest you add some words that the CH341 driver should be checked/installed BEFORE trying to load/update the code  and that there is a filesystem load as well as the code load.

**Does the filrsystem load only need to be done one time?** 

We know - **YOU** change the code, **WE** need to upload the code!

Getting my pics ready for viewing

Dave Clifford

**Comment by Alan-Lomax** [comment, marked as answer]
I am going to put a ribbon on this and call it closed. My initial issue was being unable to upload the code. It turned out to be me using a USB 2.0 cable being in between a USB3 computer port and the USB C port on the firebeetle. I am sure the two ends negotiated a high speed link that the cable in the middle could not cope with. Changing to a USB 2 port on an external USB port expander solved the issue. Note: I did not need any additional drivers for this to work (Win 10)

**Reply by Alan-Lomax** [reply]
PS: In my readings over the past couple of weeks I have come to understand **OTA** means '**Over The Air**'. Meaning code can be downloaded to the firebeetle using wifi - without getting involved in any USB cables.

**Reply by alex-code** [reply]
They do seem to be a bit temperamental with certain ports and cables.
Guess I've been lucky as I've never had any issues.

OTA could be a future feature but you'd still need to upload the code via USB first so it can handle OTA updates.

**Comment by daveclifford** [comment]
Alex,

I hope you get this file.

These are instructions to set up the tools, port permissions, etc. for
Linux users.

One of my contributions to the cause

Dave Clifford
